docs/csharp/tutorials/console-webapiclient.md

```diff
@@ -1,388 +1,354 @@
-information from GitHub.
-
-After you've configured the <xref:System.Net.Http.HttpClient>, you make a web request and retrieve the response. In
-this first version, you use the <xref:System.Net.Http.HttpClient.GetStringAsync(System.String)?displayProperty=nameWithType> convenience method. This convenience method
-starts a task that makes the web request, and then when the request returns, it reads the
-response stream and extracts the content from the stream. The body of the response is returned
-as a <xref:System.String>. The string is available when the task completes.
-
-The final two lines of this method await that task, and then print the response to the console.
-Build the app, and run it. The build warning is gone now, because the `ProcessRepositories` now
-does contain an `await` operator. You'll see a long display of JSON formatted text.
-
-## Processing the JSON Result
-
-At this point, you've written the code to retrieve a response from a web server, and display
-the text that is contained in that response. Next, let's convert that JSON response into C#
-objects.
-
-The <xref:System.Text.Json.JsonSerializer?displayProperty=nameWithType> class serializes objects to JSON and deserializes JSON into objects. Start by defining a class to represent the `repo` JSON object returned from the GitHub API:
-
-```csharp
-using System;
-
-namespace WebAPIClient
-{
-    public class Repository
-    {
-        public string name { get; set; }
-    }
-}
-```
+   ```csharp
+   private static async Task ProcessRepositories()
+   {
+       client.DefaultRequestHeaders.Accept.Clear();
+       client.DefaultRequestHeaders.Accept.Add(
+           new MediaTypeWithQualityHeaderValue("application/vnd.github.v3+json"));
+       client.DefaultRequestHeaders.Add("User-Agent", ".NET Foundation Repository Reporter");
+   
+       var stringTask = client.GetStringAsync("https://api.github.com/orgs/dotnet/repos");
+   
+       var msg = await stringTask;
+       Console.Write(msg);
+   }
+   ```
 
-Put the above code in a new file called 'repo.cs'. This version of the class represents the
-simplest path to process JSON data. The class name and the member name match the names used
-in the JSON packet, instead of following C# conventions. You'll fix that by providing some
-configuration attributes later. This class demonstrates another important feature of JSON
-serialization and deserialization: Not all the fields in the JSON packet are part of this class.
-The JSON serializer will ignore information that is not included in the class type being used.
-This feature makes it easier to create types that work with only a subset of the fields in
-the JSON packet.
+   This code:
 
-Now that you've created the type, let's deserialize it.
+   * Sets up HTTP headers for all requests: an `Accept` header to accept JSON responses, and a `User-Agent` header. These headers are checked by the GitHub server code and are necessary to retrieve information from GitHub.
```

Comment:
```suggestion
   * Sets up HTTP headers for all requests: an [`Accept`](https://developer.mozilla.org/docs/Web/HTTP/Headers/Accept) header to accept JSON responses, and a `User-Agent` header. These headers are checked by the GitHub server code and are necessary to retrieve information from GitHub.
```

Reply:
https://developer.mozilla.org/docs/Web/HTTP/Headers/User-Agent

Reply:
I'd be tempted to split up the 2 header rather than and them. It's a long sentence.